Workiva/wf-common
# HY-2651 Add Scroll Detection

JIRA >> https://jira.atl.workiva.net/browse/HY-2651
@timmccall-wf @tomconnell-wf @todbachman-wf Code Review

## Problem Summary
The behavior surrounding scroll inertia is often obnoxious at best.  Right now, when beginning a scroll, a MouseWheelStart event is fired.  This is followed by some number of MouseWheel Events (which include synthetic momentum-driven events), independent of how many scrolls are actually performed, or even if the scroll switches directions.  It would be nice if there was a way to detect when a user physically initiates a new scroll gesture.

## Solution
Implemented a solution in the MouseAdapter based on statistical analysis, looking at the recent history of scroll deltas and looking for patterns that indicate that a user performed a new scroll event.

## Unit tests
One changed, none added yet.  I'm trying to decide on a practical way to unit test this functionality.  Right now, the only option i can think of is to feed in a long lists of fake MouseEvents/deltas that should indicate a user scroll and test whether it detected it or not.  But this doesn't seem very elegant.  Additionally, i don't like the idea of such implementation dependent test cases, where slight tweaks and  changes under the abstraction hood can cause little or no functional difference, but can cause tightly fitted unit tests to fail en masse.  If reviewer(s) have any opinions and/or suggestions, they would be appreciated.

## How to QA/+10
1. bower link this branch of wf-common into wf-uicomponents and run `grunt serve` in wf-uicomponents directory.  This should automatically pop up a window for functionally testing wf-uicomponents.
2. Open the debug console and put a breakpoint at `bower_components/wf-common/src/MouseAdapter.js:320` inside the MouseAdapter constructor, after the observables are set up.  Refresh.
3. While breakpointed, attach a debugging function to the onMouseWheelStart and/or onMouseWheelEnd observables to see when they are fired.  I use `this.onMouseWheelStart(function(event) { console.log("Start event!"); }); this.onMouseWheelEnd(function(event) { console.log("End event!"); });`
4. Thoroughly test the scroll behavior, use your best judgement to determine if events are firing reasonably and consistently with your scroll gestures.  Pay particular attention to scrolling gestures you make between a cold start and debounced end-event, as those are the events this PR is all about.
